app/services/user.py

```diff
@@ -1,31 +1,86 @@
+from typing import Optional
+
+from argon2 import PasswordHasher
+
+from app.errors.user import (
+    InvalidRoleException,
+    UsernameAlreadyExistsException,
+    UsernameRequiredException,
+    UserNotFoundException,
+)
 from app.models.users import User
 from app.repositories.user import UserRepository
+from app.schemas.user import ROLE_MAPPING, ActionSuccess, RoleEnum, UserCreateSchema, UserSchema, UserUpdateSchema
 
 
 class UserService:
     def __init__(self, repository: UserRepository):
         self.repository = repository
+        self.ph = PasswordHasher()
+
+    def _get_role_id(self, role: str) -> int:
+        try:
+            role_enum = RoleEnum(role)
+            return ROLE_MAPPING[role_enum]
+        except (ValueError, KeyError):
+            raise InvalidRoleException()
 
     def get_user(self, username: str):
-        """Example method NOT FINAL for getting a user."""
+        """Getting a user."""
+
+        if not username:
+            raise UsernameRequiredException()
+        user = self.repository.get_by_username(username)
+        if not user:
+            raise UserNotFoundException()
+        return UserSchema.from_orm(user)

```

Comment:
No need to add the schema validation here, you can safely return only user
```suggestion
        return user
```